packages/protocol/contracts/identity/IdentityProxyHub.sol

```diff
@@ -0,0 +1,21 @@
+pragma solidity ^0.5.13;
+
+import "./IdentityProxy.sol";
+
+contract IdentityProxyHub {
+  mapping(bytes32 => IdentityProxy) public identityProxies;
+
+  function getIdentityProxy(bytes32 identifier) public returns (IdentityProxy) {
```

Comment:
In the view that Ethereum is a database, I feel like this should be split into two functions: A view function called `getIdentityProxy` that returns the address or `address(0)` and a state-modifying function `createIdentityProxy` that creates a new proxy of no-ops or reverts if one already exists.

Reply:
So the getter already exists - `identityProxies` is a public mapping. I think my intuition for why `getIdentityProxy` works the way it does is akin to the Singleton pattern. There's essentially supposed to be a singleton IdentityProxy per identifier, and this is the function for getting it.

Reply:
I agree that the naming here is misleading
imo any function prefixed with `get` I would expect to only use `eth_call` against